src/main/java/sim/coffee/ControlsMerchandise.java

```diff
@@ -87,18 +87,21 @@ public String getItemDetails() {
         return Merchandise.formatDetails(l, c);
     }
 
-    private void updatePrice() {
+    // Returns the price of the item given the current control values
+    public BigDecimal getPrice() {
         // Can't update price while in temporary inconsistent state
         // This occurs when items are removed from the list on population
         if (labels.getSelectedIndex() == -1) {
-            return;
+            return new BigDecimal("0");
```

Comment:
Yep this

Reply:
? This wasn't the source of the issue if that's what you mean, the change in CustomerGUI is the real issue. This line is just a result of the refactor.